HaxeManual/04-class-field.tex

```diff
@@ -286,7 +286,7 @@ \subsection{Inline}
 
 It is not always easy to judge if a function qualifies for being inline. Short functions that have no writing expressions (such as a \expr{=} assignment) are usually a good choice, but even more complex functions can be candidates. However, in some cases inlining can actually be detrimental to performance, e.g. because the compiler has to create temporary variables for complex expressions.
 
-Inline is not guaranteed to be done. The compiler might cancel inlining for various reasons or a user could supply the \ic{--no-inline} command line argument to disable inlining. The only exception is if the class is \tref{extern}{lf-externs} or if the class field has the \expr{:extern} \tref{metadata}{lf-metadata}, in which case inline is forced. If it cannot be done, the compiler emits an error.
+Inline is not guaranteed to be done. The compiler might cancel inlining for various reasons or a user could supply the \ic{--no-inline} command line argument to disable inlining. The only exception is if the class is \tref{extern}{lf-externs} or if the class field has the \tref{\expr{extern}}{class-field-extern} access modifier, in which case inline is forced. If it cannot be done, the compiler emits an error.
```

Comment:
Actually, Haxe 4 doesn't cancel inlining anymore at all. The last case where this was done was for macro-in-macro situations, but those are gone now.

Reply:
So is `extern inline` pointless / redundant now?

Reply:
What's the point of `extern inline` then?

Reply:
Hmm, I forgot about constructor inlining which is a bit of a separate case. Here the inlining might still be canceled after all and then `extern inline` causes an error. So maybe it's alright to keep this as-is.